cmd/main.go

```diff
@@ -45,7 +44,14 @@ var rootCmd = &cobra.Command{
 		var kubeClient kubernetes.Interface
 		var err error
 
-		if !refreshResources {
+		cfgManager := config.NewManager(cfgFilePath)
+		cfg, err := cfgManager.LoadConfig()
+		if err != nil {
+			fmt.Printf("Failed to load configuration file '%s': %s", cfgFilePath, err.Error())
+			os.Exit(1)
```

Comment:
OK this exit caught my eye.  I was assuming we could go with defaults for our three config fields and continue to function, with appropriate logging, if the mounted config file for the config map was missing.

So I went back to the Jira and looked at the OCM-O pointer, and tried to track down what it does.

So starting with https://github.com/openshift/cluster-openshift-controller-manager-operator/blob/master/pkg/operator/sync_openshiftcontrollermanager_v311_00.go#L183-L189

Which took me to https://github.com/openshift/cluster-openshift-controller-manager-operator/blob/master/bindata/v3.11.0/openshift-controller-manager/ds.yaml#L44-L45

Skip a few steps to 

https://github.com/openshift/openshift-controller-manager/blob/fd48f0a784383782b1ac58d95e732f8df516b844/pkg/cmd/openshift-controller-manager/cmd.go#L65-L68

I do see this:  https://github.com/openshift/openshift-controller-manager/blob/fd48f0a784383782b1ac58d95e732f8df516b844/pkg/cmd/openshift-controller-manager/cmd.go#L95-L98

So this is following the same "precedence" that was established in OCM-O etc.

I still wonder if we can go over and above, and assume defaults here too. 

What do you think @otaviof ?  Among other things, you doc file as I currently read it that we will assume defaults if the file is missing.

Lastly, also please add a TODO comment that we need to make sure how we report the lack of the config map here lines up with how our operator manages us with the.  Presumably the opeartor will also note the lack of configmap and we can decide there how it treats that with respect to reporting status to the CSI.  Some for of "degraded".  Or do we create a metric / alert for this condition perhaps.

Reply:
Have you considered what the LoadConfig does, and [how it may error out](https://github.com/otaviof/csi-driver-shared-resource/blob/04372abf4f3b05cf1e01395d64cf87207f8adf61/pkg/config/manager.go#L48-L59)?

Reply:
OK I think I have pieced together what you are getting at, and I appreciate the code links.

To state a bit more precisely and specificity:  
1) if the config file from the mounted config map is removed, assume defaults, no error returned
2) if that file is somehow corrupt, we don't want to assume defaults, since the user may have intended to override the defaults

Perhaps that should have been obvious to this reviewer.  A comment capturing 2) would have helped me, but I won't insist on it :-)

Now, as part of my original comment for a TODO comment around following up how the operator knows that this driver is giving up because the config file is corrupt is still warranted.

We can then sort out the resolution of that TODO as we sort out how much of the operator side you complete with BUILD-340.  Perhaps the TODO remains until any potential follow up to BUILD-340 is done.